src/main/java/org/spongepowered/common/mixin/core/block/MixinBlock.java

```diff
@@ -119,6 +128,15 @@ public void setTickRandomly(boolean tickRandomly) {
         this.needsRandomTick = tickRandomly;
     }
 
+    @Overwrite
+    public void randomTick(World worldIn, BlockPos pos, IBlockState state, Random random) {
+        BlockRandomTickEvent event = SpongeEventFactory.createBlockRandomTick(Sponge.getGame(), null, new Location((Extent)worldIn, VecHelper.toVector(pos)));
```

Comment:
The entire method here is rewritten, is an @Inject capable of adding a return to the injected method? If so, I could do a head inject